crates/oxc_ecmascript/src/string_substring.rs

```diff
@@ -0,0 +1,37 @@
+use crate::ToInt32;
+
+pub trait StringSubstring {
+    /// `String.prototype.substring ( start , end ] )`
+    /// <https://tc39.es/ecma262/#sec-string.prototype.substring>
+    fn substring(&self, start: Option<f64>, end: Option<f64>) -> String;
```

Comment:
Why not take a `u32` then have callers use `to_int_32` for casting?

Reply:
i'm open to change it -  this was just keeping the pat of the others https://github.com/oxc-project/oxc/blob/bbca7436890f6ef971807b7079f1472bc3e5a8df/crates/oxc_ecmascript/src/string_index_of.rs#L11

but as i'm thinking of this more, i wonder whether we need a const evaluation crate that accpts a JS `Value` or similar. accepting valid inputs really limits the minification we can do

Reply:
I'll think about these later.